core/commonMain/src/kotlinx/serialization/internal/Platform.common.kt

```diff
@@ -85,25 +85,30 @@ internal inline fun <T> SerializationStrategy<*>.cast(): SerializationStrategy<T
 
 @Suppress("UNCHECKED_CAST", "NOTHING_TO_INLINE")
 @PublishedApi
-internal inline fun <T> DeserializationStrategy<*>.cast(): DeserializationStrategy<T> = this as DeserializationStrategy<T>
+internal inline fun <T> DeserializationStrategy<*>.cast(): DeserializationStrategy<T> =
+    this as DeserializationStrategy<T>
 
 internal fun KClass<*>.serializerNotRegistered(): Nothing {
-    throw SerializationException(
-        "Serializer for class '${simpleName}' is not found.\n" +
-            "Mark the class as @Serializable or provide the serializer explicitly."
-    )
+    throw SerializationException(notRegisteredMessage())
 }
 
+internal fun KClass<*>.notRegisteredMessage(): String = "Serializer for class '${simpleName}' is not found.\n" +
+        "Please ensure that class is marked as '@Serializable' and that serialization plugin is applied.\n" +
+        "It is also possible to specify serializer explicitly with '${simpleName}.serializer(typeArgs)'"
```

Comment:
We probably should also mention external non-default serializers here (they can only be used explicitly).

Also, the same message would be shown if use `someModule.serializer()` and no serializer is registered in the module.

Reply:
I've reverted even the last line -- it looks alien and requires an additional explanation for cases like:
`It is also possible to specify serializer explicitly with 'Any.serializer(typeArgs)'`. Also, it doesn't seem that it can be reasonably explained **for a newcomer** why it is possible to specify serializer explicitly when it's not found (???).

The same story with modules -- all the phrasing and corner cases I've found are not worth it